### PR TITLE
Internal: Nested Elements - Infra fixes & tests [ED-6591]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/elements/manager.js [hunk 1]
```diff
@@ -18,38 +19,31 @@ export default class ElementsManager {
 	 * Function getElementType().
 	 *
 	 * Get element type by elType, and widgetType, if exact widgetType is not found
-	 * the method will return the element type ( WidgetBase ).
+	 * the method will return the element type.
 	 *
 	 * @param {string} elType
 	 * @param {string|false} widgetType
 	 */
 	getElementType( elType, widgetType = false ) {
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:27]: Try to use the widget as an element
```suggestion
	getElementType( elType ) {
```

File: assets/dev/js/editor/elements/manager.js [hunk 2]
```diff
@@ -18,38 +19,31 @@ export default class ElementsManager {
...
+	 * @param {ElementBase|WidgetBase} element
 	 */
 	registerElementType( element ) {
 		// Validate instanceOf `element`.
 		if ( ! ( element instanceof ElementBase ) ) {
 			throw new TypeError( 'The element argument must be an instance of ElementBase.' );
 		}
 
-		const index = element.getTypeKey();
+		const index = element instanceof WidgetBase ? element.getWidgetType() : element.getType();
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:46]:
```suggestion
		const index = element.getType();
```

File: assets/dev/js/editor/elements/manager.js [hunk 3]
```diff
@@ -18,38 +19,31 @@ export default class ElementsManager {
...
 
-		return result;
+		return this.elementTypes[ elType ];
 	}
 
 	/**
 	 * Function registerElementType().
 	 *
-	 * @param {ElementBase} element
+	 * @param {ElementBase|WidgetBase} element
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:38]:
```suggestion
	 * @param {ElementBase} element
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/manager.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js]: Handled.

File: assets/dev/js/editor/elements/manager.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js]: Handled.

File: assets/dev/js/editor/elements/manager.js (hunk 3 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js]: Handled.

File: assets/dev/js/editor/elements/manager.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js]: Handled.

File: assets/dev/js/editor/elements/manager.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js]: Handled.

File: assets/dev/js/editor/elements/manager.js (hunk 3 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js]: Handled.

**Review by matipojo — COMMENTED** [review]
Please fix the checks errors and re-create a pull request

**Pull request merged**